go/def.bzl

```diff
@@ -382,16 +287,26 @@ def go_library_impl(ctx):
     _emit_go_asm_action(ctx, src, asm_hdrs, obj)
     extra_objects += [obj]
 
-  out_lib = ctx.outputs.lib
+  lib_name = _go_importpath(ctx) + ".a"
+  out_lib = ctx.new_file(lib_name)
+  search_path = out_lib.path[:-len(lib_name)]
   gc_goopts = _gc_goopts(ctx)
-  _emit_go_compile_action(ctx, go_srcs, deps, out_lib, extra_objects, gc_goopts)
-
-  transitive_libs = set([out_lib])
-  transitive_importmap = {out_lib.path: _go_importpath(ctx)}
+  transitive_libs = depset([out_lib])
+  transitive_lib_paths = [search_path]
   for dep in deps:
-     transitive_libs += dep.transitive_go_library_object
+     transitive_libs += dep.transitive_go_libraries
      transitive_cgo_deps += dep.transitive_cgo_deps
-     transitive_importmap += dep.transitive_go_importmap
+     transitive_lib_paths += dep.transitive_go_library_paths
+
+  _emit_go_compile_action(ctx,
```

Comment:
we need it to be automatic...
I am basically incapable of either noticing or remembering white space rules